### ISSUE TITLE
Creating libiqm

### ISSUE THREAD

**Issue by schulzch** [lsalzman/iqm#14]
I really like IQM for being simple, yet some aspects are quite a pain:
1. Validation is not specified (`ofs_` may not be within file bounds)
2. Some fields are undocumented (semantics of `flags`, `num_*`, etc.)
3. How to deal with models split into geometry and animation files?

Those issues could be resolved by providing a standard implementation, that deals with:

- Iteration
- Reading (endian swapping and validation)
- Writing (endian swapping and computing some numbers)
- Endian swapping
- Optional: mesh optimization (see [AMD Tootle](http://developer.amd.com/tools-and-sdks/archive/legacy-cpu-gpu-tools/amd-tootle/))
- Optional: compression (see [OpenCTM](http://openctm.sourceforge.net/))
- Optional: file migration

I think this should be done using ANSI C to keep things simple. Any thoughts on this?

**Comment by graphitemaster** [comment]
The file format is extremely simple. I see no reason other than convenience to have this. Mesh optimization need not apply here, that is the job of the artist, the exporter already optimizes indices indexing order for cache. Compression can be done with any existing standard generic compression technique (deflate for instance.) The IQM toolkit otherwise provides you with all the information needed to write your own reader / writer.

**Comment by schulzch** [comment]
Yes and no: this is about convenience and correctness - there is virtually no reason to have independent implementations other than [getting things W3C approved](https://en.wikipedia.org/wiki/Web_SQL_Database) ;-)

**Comment by jansol** [comment]
Another reason to have a library for this would be that it might make it easier to get native blender support to avoid the exporter breaking every other release.
